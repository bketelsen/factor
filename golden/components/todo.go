package components

// Note: this entire file is autogenerated

import (
	"github.com/bketelsen/factor/golden/models"
)

// Todo is how factor binds a model to a component. It holds the an autogenerated
// TodoClient that holds a single Todo in memory and also provides a typesafe
// interface to the RPC calls to modify the TODO on the server.
//
// Don't create this struct directly, use NewTodo instead
type Todo struct {
	// this is the
	Client models.TodoClient
}

// Get is a convenience function that gets the current in-memory Todo
func (t Todo) Get() models.Todo {
	return t.Client.GetCurrent()
}

// Permalink is a convenience function to get the permalink for this todo.
func (t Todo) Permalink() string {
	return fmt.Sprintf("/todos/%s", t.Get().ID.String())
}

// NewTodo creates a new Todo with a fully populated client
func NewTodo(t models.Todo) Todo {
	return Todo{
		Client: models.NewTodoClient(t)
	}
}
