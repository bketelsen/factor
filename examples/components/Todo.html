<!-- maybe factor can somehow "bind" the models.Todo to this html? -->
<div>
    <h1>{vecty-field:Name}</h1>
    <small>{vecty-field:Description}</small>
    <div>(<a href="{vecty-field:Permalink}">Permalink</a>)</div>
    <div>Active for {vecty-call:GetAge} days</div>
</div>
