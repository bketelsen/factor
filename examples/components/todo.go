package components

import "github.com/gowasm/vecty"

type Todo struct {
	vecty.Core
	Name        string
	Description string
	Permalink   string
}

func (t *Todo) GetAge() string {
	return "5"
}
