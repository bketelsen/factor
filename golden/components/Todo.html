<!-- maybe factor can somehow "bind" the models.Todo to this html? -->
<div>
    <h1>{vecty-field:Name}</h1>
    <small>{vecty-field:Description}</small>
    <div>(<a href="{vecty-field:Permalink}">Permalink</a>)</div>
    <div>You are {vecty-call:GetAge} years old</div>
    <div>gary{vecty-call:GetSomething}susan{vecty-call:GetSomething}bob</div>
</div>
